### DISCUSSION TITLE
Random Mint function

### DISCUSSION THREAD

**Discussion by akshay-anv** [TENK-DAO/tenk#42] [Q&A] [answered]
This repo is absolutely useful  though one thing I havent been able to figure out is how to randomize the mint function meaning I minted a testnet contract 2 times and what I got was 1.png and 2.png but I wanted to randomize that process . Is it possible with a pre-defined function written in the contract?

**Comment by willemneal** [comment, marked as answer]
By default the minting is random.  There is a Raffle, which allows you to randomly pick one of the remaining token_ids (integers in some range). Perhaps you got lucky?  How big was your collection and which methods did you call?

**Reply by akshay-anv** [reply]
Yes , my sample size was very small in the testnet perhaps that was the reason . Thanks for help . Also in the start_premint function what does duration represent ?

**Reply by willemneal** [reply]
how many blocks until the pre-sale can end.  Blocks are about 1.2-1.3/sec. You can just put it to zero if you want to ignore it.

Also an update is coming that will use the `block_timestamp`, which will allow the sales to start and end automatically.
